src/main/java/de/tum/in/www1/artemis/service/plagiarism/ContinuousPlagiarismControlService.java

```diff
@@ -0,0 +1,75 @@
+package de.tum.in.www1.artemis.service.plagiarism;
+
+import org.jvnet.hk2.annotations.Service;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Profile;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import de.jplag.exceptions.ExitException;
+import de.tum.in.www1.artemis.domain.Exercise;
+import de.tum.in.www1.artemis.domain.ProgrammingExercise;
+import de.tum.in.www1.artemis.domain.TextExercise;
+import de.tum.in.www1.artemis.domain.modeling.ModelingExercise;
+import de.tum.in.www1.artemis.repository.ExerciseRepository;
+import de.tum.in.www1.artemis.service.util.TimeLogUtil;
+
+/**
+ * Manages continuous plagiarism control.
+ */
+@Service
+@Component
+@Profile("scheduling")
+public class ContinuousPlagiarismControlService {
+
+    private static final Logger log = LoggerFactory.getLogger(ContinuousPlagiarismControlService.class);
+
+    private final ExerciseRepository exerciseRepository;
+
+    private final PlagiarismChecksService plagiarismChecksService;
+
+    public ContinuousPlagiarismControlService(ExerciseRepository exerciseRepository, PlagiarismChecksService plagiarismChecksService) {
+        this.exerciseRepository = exerciseRepository;
+        this.plagiarismChecksService = plagiarismChecksService;
+    }
+
+    /**
+     * Daily triggers plagiarism checks as a part of continuous plagiarism control.
+     */
+    @Scheduled(cron = "0 0 4 * * *") // execute this every night at 4:00:00 am
```

Comment:
Have you considered making this configurable, similarly as it is done in #6630 for the cron expressions.

Reply:
That's good point

Reply:
Done

Reply:
before we merge this, we have to add the property to the ansible collection. https://github.com/ls1intum/artemis-ansible-collection. You can just open a PR there.